ocean_provider/validation/requests.py

```diff
@@ -48,6 +68,103 @@ def __init__(
         self._processor = CustomRulesProcessor()
 
 
+class RBACValidator:
+    def __init__(self, request_name=None, request=None, assets: Optional[list] = None):
+        self._request = request
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+        self._action = action_mapping[request_name]
+        self._credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self._component = "provider"
+        self._assets = assets
+
+    @property
+    def credentials(self):
+        return self._credentials
+
+    @property
+    def component(self):
+        return self._component
+
+    @property
+    def action(self):
+        return self._action
+
+    @property
+    def assets(self):
+        return self._assets
+
+    @property
+    def request(self):
+        return self._request
+
+    def passes(self):
+        return True
+
+    def fails(self):
```

Comment:
this needs to be adapted. Please remove the passes function, since it is not needed at all. And implement the contents for the fails function. This is now only a placeholder.